File: Claude/Python/2025/Day1_Solution_2025.py
import time

start = time.perf_counter()

with open("Day1_Input.txt") as f:
    rotations = f.read().split()

pos = 50
count = 0

for rot in rotations:
    direction = rot[0]
    dist = int(rot[1:])
    if direction == 'L':
        pos = (pos - dist) % 100
    else:
        pos = (pos + dist) % 100
    if pos == 0:
        count += 1

print(count)

end = time.perf_counter()
print(f"Runtime: {end - start:.6f} seconds")

File: Claude/Rust/2025/Day1_Solution_2025.rs
use std::fs;

fn main() {
    let content = fs::read_to_string("Day1_Input.txt").expect("Could not read input file");

    let mut position: i64 = 50;
    let mut count = 0u64;

    for line in content.lines() {
        let line = line.trim();
        if line.is_empty() {
            continue;
        }
        let (dir, dist) = line.split_at(1);
        let dist: i64 = dist.parse().expect("Invalid distance");
        position = match dir {
            "R" => (position + dist).rem_euclid(100),
            "L" => (position - dist).rem_euclid(100),
            _ => panic!("Unknown direction"),
        };
        if position == 0 {
            count += 1;
        }
    }

    println!("{}", count);
}
